tests/pytest/core/test_views.py

```diff
@@ -183,12 +170,12 @@ def test_help_with_session_agency(client):
 
 
 @pytest.mark.django_db
-def test_bad_request_active_agency(app_request, mocked_active_agency, mocked_session_update):
+def test_bad_request_active_agency(app_request, mocked_session_agency, mocked_session_update):
     response = bad_request(app_request, Exception())
 
     assert response.status_code == 400
     assert "origin" in mocked_session_update.call_args.kwargs
-    assert mocked_session_update.call_args.kwargs["origin"] == mocked_active_agency.index_url
+    assert mocked_session_update.call_args.kwargs["origin"] == mocked_session_agency.return_value.index_url
```

Comment:
I thought about this refactor and looked at the view / middleware code for a good bit. I agree with the conclusion that the fixture is redundant :+1:

Reply:
See the commit message on that for more of an explanation: c53e4d5be12ca4285bd46a44a440ea7d28a76f5a